src/main/java/jdbchomework/dao/hibernate/AbstractHibDao.java

```diff
@@ -11,58 +11,62 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public abstract class AbstractDao<T extends AbstractEntity> implements GenericDao<T> {
 
+public class AbstractHibDao<T extends AbstractEntity> implements GenericDao<T> {
     public static final String CANNOT_CONNECT_TO_DB = "Cannot connect to DB";
-    private org.slf4j.Logger log = LoggerFactory.getLogger(AbstractDao.class);
-    private SessionFactory sessionFactory;
-    private String entity;
+    private org.slf4j.Logger log = LoggerFactory.getLogger(AbstractHibDao.class);
+    protected SessionFactory sessionFactory;
+    private String entityName;
     private Class<T> clazz;
 
-    public AbstractDao(SessionFactory sessionFactory, String entity, Class<T> clazz) {
+    public AbstractHibDao(String entityName, SessionFactory sessionFactory, Class<T> clazz) {
+        this.entityName = entityName;
         this.sessionFactory = sessionFactory;
-        this.entity = entity;
         this.clazz = clazz;
+
     }
 
+    @Override
     public void add(T toAdd) {
-        Session session = getSession();
         Transaction transaction = null;
-        try {
+
+        try (Session session = sessionFactory.openSession()) {
             transaction = session.beginTransaction();
             session.save(toAdd);
             transaction.commit();
         } catch (HibernateException e) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            e.printStackTrace();
+            log.error(CANNOT_CONNECT_TO_DB, e);
+            throw new RuntimeException(CANNOT_CONNECT_TO_DB, e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)